CodePush.ios.js

```diff
@@ -210,7 +209,13 @@ function sync(options = {}, syncStatusChangeCallback, downloadProgressCallback)
           resolve(CodePush.SyncStatus.UP_TO_DATE);
         }
         else if (syncOptions.updateDialog) {
-          syncOptions.updateDialog = Object.assign(CodePush.DEFAULT_UPDATE_DIALOG, syncOptions.updateDialog);
+          // updateDialog supports any truthy value (e.g. true, "goo", 12),
+          // but we should treat a non-object value as just the default dialog
+          if (typeof syncOptions.updateDialog !== "object") {
+            syncOptions.updateDialog = CodePush.DEFAULT_UPDATE_DIALOG;
+          } else {
+            syncOptions.updateDialog = Object.assign(CodePush.DEFAULT_UPDATE_DIALOG, syncOptions.updateDialog);
```

Comment:
checkForUpdate() or sync()? the former doesn't use CodePush.DEFAULT_UPDATE_DIALOG right?